api/handlers.go

```diff
@@ -170,16 +170,16 @@ func (a *HttpApi) RetrieveQueue(w http.ResponseWriter, r *http.Request) {
 		queue, err := a.storage.RetrieveQueue(uint(queueID))
 
 		if err != nil {
-			write(w, http.StatusNotFound, ErrorResponse{
+			Write(w, http.StatusNotFound, ErrorResponse{
 				Message: fmt.Sprintf("Queue with ID %d not found", queueID),
 				Status:  http.StatusNotFound,
 			})
 		} else {
 			pendingTasks := a.storage.RetrieveTasksByState(queue.ID, storage.TaskPending)
 			runningTasks := a.storage.RetrieveTasksByState(queue.ID, storage.TaskRunning)
-			response := responseFromQueue(queue, uint(len(pendingTasks)), uint(len(runningTasks)))
+			response := responseFromQueue(queue, uint(len(pendingTasks)), uint(len(runningTasks)), uint(len(queue.Workers)))
```

Comment:
why do we have all these concepts of pending, running tasks already in the code? how is this related to the join feature? is this an old codebase?

Reply:
Yes, it is from the old codebase.